Automattic/wp-calypso
# Add e2e tests to the FSE plugin

_Note: split from  #39000_

### Changes proposed in this Pull Request
* Uses @wordpress/scripts to run the e2e test commands.
* Run e2e tests against default environment URL
* Adds @wordpress/e2e-test-utils to help with e2e testing
* Adds an e2e test which tests to see if our back button override exists
* Adds a place for e2e helper functions, including one for activating a theme

### e2e tests
The e2e command automatically runs test files that:
1. Are in `apps/full-site-editing`
2. Have the extension `.spec.ts` or `.spec.js` OR is a js/ts file within a `specs` folder

### Testing instructions
1. Create and start a WordPress testing environment using #39000. (port must be localhost:4013)
2. Run `npm ci` from wp-calypso root
2. Run `npx lerna run build --scope='@automattic/full-site-editing' --stream --no-prefix` from wp-calypso root
2. Run `npx lerna run test:e2e --scope='@automattic/full-site-editing' --stream --no-prefix` from wp-calypso root
3. Tests should pass successfully

To run the e2e tests:
```sh
npx lerna run test:e2e --scope='@automattic/full-site-editing' --stream
```
